ortools/linear_solver/MPSWriteError.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2023 RTE
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Initial version of this code was provided by RTE
+
+#ifndef ORTOOLS_MPSWRITEERROR_H
+#define ORTOOLS_MPSWRITEERROR_H
+#include <stdexcept>
+namespace operations_research {
+class MPSWriteError : public std::runtime_error {
+ public:
+  MPSWriteError(std::string message) : std::runtime_error(message) {}
```

Comment:
Log message here ?

```suggestion
  MPSWriteError(const std::string& message) : std::runtime_error(message) {}
```

Reply:
since we're throwing an exception, I think we shouldn't log it.

Reply:
If the user doesn't write something like
```cpp
try {
solver->Write(...);
} catch (const std::runtime_error& ex) {
// ...
}
```
The behavior will be an error message printed in `std::cerr` such as

> terminate called after throwing an instance of 'MPSWriteError'
> Abandon (core dumped)

If you think this is enough, we can keep it like this. Keep in mind that in some cases, `std::cerr` may not be captured, only logs.

Reply:
I do think that it's up to the user in such a case to catch & log the exception. But if you strongly feel against it, we can add the log

Reply:
OK, we can keep it like this. But then we have to document this (`Write` function may throw XXX)